File: FileDriveServer/FileDriveWebAPI/Models/Permission.cs
using FileDriveWebAPI.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace FileDriveWebAPI.Models
{
    public class Permission
    {
        public int Id { get; set; }
        public User User { get; set; }
        public TreeEntity Entity { get; set; } 
        public ENUMPermissionType PermissionType { get; set; }
    }
}

File: file-drive-client/src/models/Permission.ts
import { ENUMPermissionType } from './../enums/ENUMPermissionType';
import { User } from './../models/User';
import { ITreeEntity } from './../files-tree/interfaces/ITreeEntity';

export class Permission {
    public id: number
    public user: User
    public entity: ITreeEntity
    public permissionType: ENUMPermissionType


    public constructor(id: number, user: User, entity: ITreeEntity, permissionType: ENUMPermissionType) {
        this.id = id;
        this.user = user;
        this.entity = entity;
        this.permissionType = permissionType;
    }
}